Shopify/shopify-cli
# Enable Theme Kit Access passwords

### WHY are these changes introduced?

First step for https://github.com/Shopify/shopify-cli-planning/issues/332
Fixes https://github.com/Shopify/shopify-cli-planning/issues/365

### WHAT is this pull request doing?

- Enable Theme Kit Access passwords through env variable. When `SHOPIFY_CLI_ADMIN_AUTH_TOKEN` contains a Theme Kit Access password, the theme requests will go through Theme Kit Access API instead of Admin API, so the password is validated. As the login is not required, it also requires passing the store URL through `SHOPIFY_SHOP`.
- Rename `SHOPIFY_CLI_STORE` to `SHOPIFY_SHOP`, which was already existing. It was [added recently](https://github.com/Shopify/shopify-cli/pull/2269) to be used from the CLI3. I'll update the CLI3 as well to use `SHOPIFY_SHOP`.

### How to test your changes?

Check that theme commands work the same way when logged in (by using the Admin API):
- `bin/shopify login`
- `DEBUG=1 bin/shopify theme list`

Check that they also work with a Theme Kit Access password (by using Theme Access Kit API):
- Install [Theme Kit Access](https://apps.shopify.com/theme-kit-access) app and generate a new password
- `bin/shopify logout`
- `DEBUG=1 SHOPIFY_SHOP=your-shop.myshopify.com SHOPIFY_CLI_ADMIN_AUTH_TOKEN=your-password bin/shopify theme list`

### Update checklist
